### PR TITLE
feat/WIT-112-hangman-command

### PR DESCRIPTION
**Pull request by katejoh**
- Implement a hangman game
- Runs using the command '\play-hangman', where a random word is generated and users can guess letters using buttons

### CODE REVIEW

File: cogs/hangman.py [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+            )
+            button.callback = button_callback
+            self.buttons.append(button)
+
+    def split_buttons_into_pages(self):
+        self.pages = [
+            self.buttons[i : i + self.page_size]
+            for i in range(0, len(self.buttons), self.page_size)
+        ]
+
```
Comment (aolinxuu) [inline cogs/hangman.py:184]: Possibly add a comment here to describe the functionality of create_button_callback

File: cogs/hangman.py [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+        self.incorrectLetters = []
+        self.pages = []  # list to store button pages
+        self.page_size = 13  # Number of buttons per page
+        self.category = category
+        self.create_buttons()
+        self.split_buttons_into_pages()
+        self.current_page = 0
+        self.add_buttons_to_current_page()
+        self.add_page_switching_buttons()
+        self.hangmanStages = [
```
Comment (aolinxuu) [inline cogs/hangman.py:110]: change to hangman_stages

File: cogs/hangman.py [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+        self,
+        inter: discord.Interaction,
+        category: app_commands.Choice[int],
+        word_length: str,
+    ):
+        await inter.response.defer()
+        r = RandomWord()
+
+        if category.value == 1:
+            selectedCategory = ["nouns"]
```
Comment (aolinxuu) [inline cogs/hangman.py:35]: change to snake case for all selectedCategory

File: cogs/hangman.py [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+    )
+    @app_commands.choices(
+        category=[
+            app_commands.Choice(name="Noun", value=1),
+            app_commands.Choice(name="Adjective", value=2),
+            app_commands.Choice(name="Verb", value=3),
+            app_commands.Choice(name="Random", value=4),
+        ],
+    )
+    async def playHangman(
```
Comment (aolinxuu) [inline cogs/hangman.py:25]: change to snake case

File: cogs/hangman.py [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+    def guess_is_correct(self, guess):
+        correct = guess in self.randomWord
+        if correct:
+            self.revealedWord = "".join(
+                c if c == guess else r
+                for c, r in zip(self.randomWord, self.revealedWord)
+            )
+        return correct
+
+    async def updateMessage(self, interaction, button_id):
```
Comment (aolinxuu) [inline cogs/hangman.py:210]: change to snake case

File: cogs/hangman.py [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+        # Create a new embed with the updated information
+        incorrectLettersString = (
```
Comment (aolinxuu) [inline cogs/hangman.py:212]: snake case

File: cogs/hangman.py [hunk 7]
```diff
@@ -0,0 +1,293 @@
...
+        )
+
+
+class MyView(discord.ui.View):
+    def __init__(self, creatorId, randomWord, category):
+        super().__init__(timeout=None)
+        self.creatorId = creatorId
+        self.randomWord = randomWord.upper()
+        self.revealedWord = "".join("_" for c in self.randomWord)
+        self.incorrectLetters = []
```
Comment (aolinxuu) [inline cogs/hangman.py:101]: snake case

**Review by aolinxuu — COMMENTED** [review]
Overall, it's a well-implemented and clean piece of code with minor improvements possible in terms of adding comments for some methods and also looking over variables and methods to make sure they follow snake case.

File: cogs/hangman.py (hunk 2 above)
Comment (katejoh) [reply cogs/hangman.py:110]: Resolving as the project currently uses camelCase.

File: cogs/hangman.py (hunk 3 above)
Comment (katejoh) [reply cogs/hangman.py:35]: Resolving as the project currently uses camelCase.

File: cogs/hangman.py (hunk 5 above)
Comment (katejoh) [reply cogs/hangman.py:210]: Resolving as the project currently uses camelCase.

File: cogs/hangman.py (hunk 4 above)
Comment (katejoh) [reply cogs/hangman.py:25]: Resolving as the project currently uses camelCase.

File: cogs/hangman.py (hunk 6 above)
Comment (katejoh) [reply cogs/hangman.py:212]: Resolving as the project currently uses camelCase.

File: cogs/hangman.py (hunk 7 above)
Comment (katejoh) [reply cogs/hangman.py:101]: Resolving as the project currently uses camelCase.

**Pull request merged**